lib/resourcemerge/batch.go

```diff
@@ -2,21 +2,97 @@ package resourcemerge
 
 import (
 	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/utils/pointer"
 )
 
 // EnsureJob ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func EnsureJob(modified *bool, existing *batchv1.Job, required batchv1.Job) {
-
 	if required.Spec.Selector != nil {
 		panic("spec.selector is not supported in Job manifests")
```

Comment:
Why doesn't this move down into `ensureJobSpec`?

Reply:
Nice catch, I didn't notice that. I will move it to the `ensureJobSpec`.